base/workerpool.jl

```diff
@@ -114,3 +117,57 @@ using `remotecall_fetch`.
 """
 remote(f) = (args...; kwargs...)->remotecall_fetch(f, default_worker_pool(), args...; kwargs...)
 remote(p::WorkerPool, f) = (args...; kwargs...)->remotecall_fetch(f, p, args...; kwargs...)
+
+type CachedWorkerPool # Manages a cache for code/data on workers.
+    pool::WorkerPool
+    directory::Dict{Tuple, RemoteChannel}  # Mapping between a tuple (worker_id, object_id) and a remote_ref
+    CachedWorkerPool(pool) = new(pool, Dict{Int, Any}())
+end
+
+function release(p::CachedWorkerPool)  # For a quick release of remote function cache
```

Comment:
I feel like since it is doing the job of a finalizer -- freeing resources (AFAICT), it should for semantic clarity be a finaliser -- even if it is called manually with a call to `finalize`.